src/legacy/core_plugins/vis_type_table/public/components/table_vis_options.tsx

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useEffect, useMemo } from 'react';
+import { get } from 'lodash';
+import { EuiPanel, EuiTitle, EuiSpacer } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import { tabifyGetColumns } from 'ui/agg_response/tabify/_get_columns';
+import { VisOptionsProps } from 'ui/vis/editors/default';
+import {
+  NumberInputOption,
+  SwitchOption,
+  SelectOption,
+} from '../../../kbn_vislib_vis_types/public/components/common';
+import { TableVisParams } from '../types';
+import { totalAggregations, isNumeric } from './utils';
+
+function TableOptions({
+  aggs,
+  aggsLabels,
+  stateParams,
+  setValidity,
+  setValue,
+}: VisOptionsProps<TableVisParams>) {
+  const noCol = useMemo(
```

Comment:
I think it's redundant here. It will not boost the performance, but makes this component less readable.
Just move this value inside `percentageColumns` calculation as it is.